### PR TITLE
feat: introducing structured-data audit result type

### PR DESCRIPTION
**Pull request by dzehnder**
In context of [SITES-24087](https://jira.corp.adobe.com/browse/SITES-24087), introducing URL inspection result to the audit result

### CODE REVIEW

File: docs/openapi/schemas.yaml [hunk 1]
```diff
@@ -26,6 +26,7 @@ AuditType:
     - 'broken-backlinks'
     - 'canonical'
     - 'sitemap'
+    - 'url-inspect'
```
Comment (iuliag) [inline docs/openapi/schemas.yaml:29]: This is very GSC API specific. For the audit name, we should opt for a more generic name, something along the lines of `URL indexability`.

File: docs/openapi/schemas.yaml [hunk 2]
```diff
@@ -426,6 +427,12 @@ AuditResult:
           description: A list of experimentation opportunities
           items:
             $ref: './schemas.yaml#/ExperimentationOpportunity'
+    - type: object
+      properties:
+        urlInspection:
```
Comment (iuliag) [inline docs/openapi/schemas.yaml:432]: This wrapper could be skipped and you could directly include: `index(ability)`, `richResults` with their properties.

File: docs/openapi/schemas.yaml [hunk 3]
```diff
@@ -474,6 +481,71 @@ ExperimentationOpportunity:
...
+  type: object
+  readOnly: true
+  properties:
+    indexStatusResult:
+      type: object
+      description: Result of the index status analysis.
+      properties:
+        verdict:
+          $ref: '#/Verdict'
+        lastCrawlTime:
```
Comment (iuliag) [inline docs/openapi/schemas.yaml:494]: There's a DataType already defined in schemas: https://github.com/adobe/spacecat-api-service/blob/main/docs/openapi/schemas.yaml#L6

**Review by iuliag — COMMENTED** [review]
AFAIR, we decided to go with having the audit type distinguish which audit result schema applies and this would be implemented via Mapping Type Names as this doc shows: https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/

**Review by ekremney — APPROVED** [review]

**Pull request merged**